packages/dds/map/package.json

```diff
@@ -98,5 +99,13 @@
     "rimraf": "^2.6.2",
     "typescript": "~4.1.3",
     "typescript-formatter": "7.1.0"
+  },
+  "typeValidation": {
+    "version": "0.58.3000",
+    "broken": {
+      "0.58.1001":{
+        "ClassDeclaration_SharedDirectory": {"forwardCompat": false}
```

Comment:
@jatgarg @markfields 
```
@fluidframework/map: src/test/types/validateMapPrevious.ts:411:5 - error TS2345: Argument of type 'TypeOnly<import("D:/code/fluid/node_modules/@fluidframework/map-previous/dist/directory").SharedDirectory>' is not assignable to parameter of type 'TypeOnly<import("D:/code/fluid/packages/dds/map/dist/directory").SharedDirectory>'.
@fluidframework/map:   Type 'TypeOnly<SharedDirectory>' is missing the following properties from type 'TypeOnly<SharedDirectory>': dispose, disposed
```

Reply:
maybe this is ok. like @markfields  was talking about. it only breaks forward compat, but consumer probably don't have their own implementations

Reply:
how come this is against 58.1001 instead of 58.2001?
i think this issue in particular was introduced in 58.2000 and corrected in 58.2001

Reply:
i had to do some manual fix ups in this area, as the previous type validation version was not set. must have set this incorrectly.

Reply:
hopefully this can prevent accidental break like this from slipping in